shared/middleware/helpers/storage_id.rb

```diff
@@ -63,7 +63,12 @@ def storage_encrypt_app_id(storage_id, app_id)
 
 def storage_id(endpoint)
```

Comment:
proposal: extract and rename a version of this function which only returns the access type as 'shared', 'readonly', or 'private' (exact names up for debate). re